packages/eslint-plugin/src/rules/no-floating-promises.ts

```diff
@@ -80,6 +91,20 @@ export default util.createRule<Options, MessageId>({
       },
     };
 
+    function isAsyncIife(
+      node: TSESTree.ExpressionStatement | TSESTree.ArrowFunctionExpression,
```

Comment:
```suggestion
      node: TSESTree.ExpressionStatement,
```

Reply:
`ExpressionStatement` doesnt have `ArrowFunctionExpression` expression type so its causing no overlap issue

![image](https://user-images.githubusercontent.com/26347874/77785939-f2d00580-7082-11ea-9f2b-9910e9b429e4.png)

Reply:
That's just an incorrect type definition.
You can add it to the type definition here
https://github.com/typescript-eslint/typescript-eslint/blob/133f622f38a286eac45288a9789d2ee529d5e582/packages/typescript-estree/src/ts-estree/ts-estree.ts#L384-L396